whist/server/web_socket/entry.py

```diff
@@ -0,0 +1,51 @@
+"""Routes of the websocket communication."""
+from fastapi import APIRouter, Depends, WebSocket
+from whist.core.error.table_error import PlayerNotJoinedError
+
+from whist.server.services.authentication import get_current_user
+from whist.server.services.error import GameNotFoundError
+from whist.server.services.game_db_service import GameDatabaseService
+from whist.server.services.user_db_service import UserDatabaseService
+from whist.server.web_socket.subscriber import Subscriber
+
+router = APIRouter()
+
+
+@router.websocket('/ping')
+async def ping(websocket: WebSocket):
+    """
+    Sends 'pong' to the client as response.
+    :param websocket: communication end point of the client
+    """
+    await websocket.accept()
+    await websocket.send_text('pong')
+
+
+@router.websocket('/room/{room_id}')
+async def subscribe_room(websocket: WebSocket, room_id: str,
+                         game_service: GameDatabaseService = Depends(GameDatabaseService),
+                         user_service: UserDatabaseService = Depends(UserDatabaseService)):
+    """
+    Clients requests to subscribe to room's side channel.
+    :param websocket: communication end point of the client. The body of the request must contain a
+    json object with the key 'token' with its value the bare string token.
+    :param room_id: ID of the room to which should be subscribed
+    :param game_service: handles all request to the db regarding rooms.
+    :param user_service: handles all request to the db regarding users.
+    :return: Sends response to the websockets. No return.
+    """
+    await websocket.accept()
+    try:
+        token = await websocket.receive_json()
+        player = await get_current_user(token['token'], user_service)
+        subscriber = Subscriber(websocket)
+        room = game_service.get(room_id)
+        _ = room.get_player(player)
+        room.side_channel.attach(subscriber)
+        await websocket.send_text('200')
```

Comment:
don't we need some kind of while loop here to keep the ws connection alive?
have you tested if the connection is kept alive in production?

Reply:
This is hard to test with the current setup. We should implement some notifications from the server to verify behavior.